inlong-sort/sort-connectors/jdbc/src/main/java/org/apache/inlong/sort/jdbc/internal/JdbcMultiBatchingOutputFormat.java

```diff
@@ -0,0 +1,605 @@
+            } catch (Exception e) {
+                LOG.error("unable to open JDBC writer, tableIdentifier:{} err:", tableIdentifier, e);
+                return null;
+            }
+            connectionExecProviderMap.put(tableIdentifier, tableConnectionProvider);
+            jdbcExec.prepareStatements(tableConnectionProvider.getConnection());
+        } catch (Exception e) {
+            return null;
+        }
+        jdbcExecMap.put(tableIdentifier, jdbcExec);
+        return jdbcExec;
+    }
+
+    private String getTbNameFromIdentifier(String tableIdentifier) {
+        String[] fileArray = tableIdentifier.split("\\.");
+        if (2 == fileArray.length) {
+            return fileArray[1];
+        }
+        if (3 == fileArray.length) {
+            return fileArray[1] + "." + fileArray[2];
+        }
+        return null;
+    }
+
+    private String getTDbNameFromIdentifier(String tableIdentifier) {
+        String[] fileArray = tableIdentifier.split("\\.");
+        return fileArray[0];
+    }
+
+    private void checkFlushException() {
+        if (schemaUpdateExceptionPolicy.equals(SchemaUpdateExceptionPolicy.THROW_WITH_STOP)
+                && !tableExceptionMap.isEmpty()) {
+            String tableErr = "Writing table get failed, tables are:";
+            for (Map.Entry<String, Exception> entry : tableExceptionMap.entrySet()) {
+                LOG.error("Writing table:{} get err:{}", entry.getKey(), entry.getValue().getMessage());
+                tableErr = tableErr + entry.getKey() + ",";
+            }
+            throw new RuntimeException(tableErr.substring(0, tableErr.length() - 1));
+        }
+    }
+
+    @Override
+    public final synchronized void writeRecord(In row) throws IOException {
+        checkFlushException();
+        JsonDynamicSchemaFormat jsonDynamicSchemaFormat =
+                (JsonDynamicSchemaFormat) DynamicSchemaFormatFactory.getFormat(sinkMultipleFormat);
+        if (row instanceof RowData) {
+            RowData rowData = (RowData) row;
+            JsonNode rootNode = jsonDynamicSchemaFormat.deserialize(rowData.getBinary(0));
+            String tableIdentifier = null;
+            try {
+                if (StringUtils.isBlank(schemaPattern)) {
+                    tableIdentifier = StringUtils.join(
+                            jsonDynamicSchemaFormat.parse(rootNode, databasePattern), ".",
+                            jsonDynamicSchemaFormat.parse(rootNode, tablePattern));
+                } else {
+                    tableIdentifier = StringUtils.join(
+                            jsonDynamicSchemaFormat.parse(rootNode, databasePattern), ".",
+                            jsonDynamicSchemaFormat.parse(rootNode, schemaPattern), ".",
+                            jsonDynamicSchemaFormat.parse(rootNode, tablePattern));
+                }
+            } catch (Exception e) {
+                LOG.info("Cal tableIdentifier get Exception:", e);
```

Comment:
Is there a strategy to choose instead of just ignoring errors？

Reply:
nothing design now